src/Controller/FilesController.php

```diff
@@ -78,16 +78,21 @@ public function fileAction(Request $request, string $file): BinaryFileResponse
         $allowLogin = false;
         $allowAccess = false;
 
+        $objUser = FrontendUser::getInstance();
+
         do {
             // Only check for folders and when member groups have been set
-            if ('folder' === $filesModel->type && null !== $filesModel->groups) {
+            // or access to member home directory
+            if ('folder' === $filesModel->type
+                && (null !== $filesModel->groups
+                || ($objUser->accessHomeDir && $objUser->homeDir))) {
                 $allowLogin = true;
 
                 // Set the model to protected on the fly
                 $filesModel->protected = true;
 
-                // Check access
-                if (Controller::isVisibleElement($filesModel)) {
+                // Check access member group or home directory.
+                if (Controller::isVisibleElement($filesModel) || ($objUser->homeDir ===  $filesModel->uuid)) {
```

Comment:
This would allow the user to access their home directory even when `accessHomeDir` was not enabled, if any permission was set on the home directory.

In total I would introduce the following before the loop:

```php
// Get the current user
$user = $this->security->getUser();

// Check if the current user can access their home directory
$canAccessHomeDir = $user instanceof FrontendUser && !empty($user->homeDir) && $user->accessHomeDir;
```

Then replace `$objUser->accessHomeDir && $objUser->homeDir` with `$canAccessHomeDir` in the first condition in the loop. And then:

```suggestion
                // Check if this is the home directory
                $isHomeDir = $user instanceof FrontendUser && $user->homeDir === $filesModel->uuid;

                // Check access
                if (($canAccessHomeDir && $isHomeDir) || (Controller::isVisibleElement($filesModel)) {
```